### PR TITLE
chore: improved release workflow and changelog

### PR DESCRIPTION
**Pull request by robertsLando**
@chrisns I'm thinking about switch to a new release flow. I need feedbacks about it

### CODE REVIEW

**Review by chrisns — APPROVED** [review]

**Review by chrisns — CHANGES_REQUESTED** [review]
I'm really really uncomfortable with logic gates that control use/access to privileged tokens not even having unit tests around them.

**Comment by robertsLando** [comment]
@chrisns Could you exmplain better? What should I add to make this secure?

File: .github/workflows/release-it.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  workflow_dispatch:
+    inputs:
+      type:
+        description: "Type. Can be `patch` `minor` or `major`"
+        required: true
+        default: "patch"
+
+jobs:
+  # #########################################################################
+  # Create a new release when an authorized person posts "@zwave-js-bot release <patch|minor|major>"
```
Comment (AlCalzone) [inline .github/workflows/release-it.yml:13]: The comment is no longer correct

File: .github/workflows/release-it.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+name: "Release-it: Create a new release on demand"
+
+on:
+  workflow_dispatch:
+    inputs:
+      type:
+        description: "Type. Can be `patch` `minor` or `major`"
```
Comment (AlCalzone) [inline .github/workflows/release-it.yml]: I don't see where this is used?

Comment (AlCalzone) [reply .github/workflows/release-it.yml]: And do you need to verify the input somehow?

**Review by AlCalzone — CHANGES_REQUESTED** [review]

File: .github/workflows/release-it.yml (hunk 1 above)
Comment (robertsLando) [reply .github/workflows/release-it.yml:13]: Fixed :)

File: .github/workflows/release-it.yml (hunk 2 above)
Comment (robertsLando) [reply .github/workflows/release-it.yml]: That's automatically done by release-it. Also it automatically do tests, lint and all the other stuff needed

File: .github/workflows/release-it.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+on:
+  workflow_dispatch:
+    inputs:
+      type:
+        description: 'Type. Can be `patch` `minor` or `major`'
+        required: true
+        default: 'patch'
+jobs:
+  release-it:
```
Comment (AlCalzone) [inline .github/workflows/release-it.yml:14]: You should limit this action to whoever you're comfortable with to do releases.

> To trigger a workflow in a repository, the user should be a collaborator with Write permission in the repository. Normally the external users can’t trigger workflows in the repository.

**Review by AlCalzone — CHANGES_REQUESTED** [review]

**Review by chrisns — APPROVED** [review]

File: .github/workflows/release-it.yml (hunk 3 above)
Comment (robertsLando) [reply .github/workflows/release-it.yml:14]: Fixed! Thanks :)

File: .github/workflows/release-it.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    if: contains('["robertsLando","chrisns"]', github.actor)
```
Comment (AlCalzone) [inline .github/workflows/release-it.yml:15]: I'm not sure if you even need that strange array syntax - it is a string after all ;)

File: .github/workflows/release-it.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# #########################################################################
+# Creates a new release using `workflow_dispatch` event trigger with `type`
+# as input to describe the type of release to create
+name: 'Release-it: Create a new release on demand'
+
+on:
+  workflow_dispatch:
+    inputs:
+      type:
+        description: 'Type. Can be `patch` `minor` or `major`'
```
Comment (AlCalzone) [inline .github/workflows/release-it.yml:10]: Do you need verification that the type was entered correctly?

Comment (robertsLando) [reply .github/workflows/release-it.yml:10]: Nope that's handled by release-it :)

File: .github/workflows/release-it.yml (hunk 4 above)
Comment (robertsLando) [reply .github/workflows/release-it.yml]: Dunno how I missed that but good catch! :) Fixed

File: .github/workflows/release-it.yml [hunk 6]
```diff
@@ -12,7 +12,7 @@ on:
         default: 'patch'
 jobs:
   release-it:
-    if: contains('["robertsLando","chrisns"]', github.actor)
+    if: contains(["robertsLando","chrisns"], github.actor)
```
Comment (AlCalzone) [inline .github/workflows/release-it.yml]: And now I'm not sure if that works. Either
```
if: contains("robertsLando,chrisns", github.actor)
```
which is probably unsafe if someone decides to call himself "robert", or with `fromJSON`:
https://github.community/t/passing-an-array-literal-to-contains-function-causes-syntax-error/17213/3

Comment (robertsLando) [reply .github/workflows/release-it.yml]: Used the safer way: `    if: github.actor == 'robertsLando' || github.actor === 'chrisns'`

**Review by AlCalzone — APPROVED** [review]

**Pull request merged**